### PR TITLE
Complete Junit5 migration

### PR DESCRIPTION
**Pull request by dimitarp**
* fixed imports
* fixed changed assertEquals signature (in JUnit5 the message is the last argument)
* refactored parameterized tests (JUnit4 Parameterized-Rule is no longer available)
* added missing JUnit dependencies and set a newer version of Junit5
* added missing volatile in the Job class
* some other improvements in test code
:)

### CODE REVIEW

**Comment by ChristianGruen** [comment]
Awesome… Thanks a lot for the effort, Dimitar!

File: basex-core/src/test/java/org/basex/server/SessionTest.java [hunk 1]
```diff
@@ -545,27 +539,25 @@ public void queryBind2() throws IOException {
...
+  public void queryError2() {
+    assertThrows(BaseXException.class, () -> session.query("(1,'a')[. eq 1]").execute());
   }
 
-  /** Runs an erroneous query.
-   * @throws IOException expected exception*/
-  @Test(expected = BaseXException.class)
+  /** Runs an erroneous query. */
+  @Test
+  // FIXME: it seems that an error is thrown on the first next() call even though it should be ok
```
Comment (ChristianGruen) [inline basex-core/src/test/java/org/basex/server/SessionTest.java:556]: @dimitarp The test doesn’t trigger an error on my system. How can this be reproduced?

Comment (ChristianGruen) [reply basex-core/src/test/java/org/basex/server/SessionTest.java:556]: …and I was too quick. I’ll have a look into it.

Comment (ChristianGruen) [reply basex-core/src/test/java/org/basex/server/SessionTest.java:556]: Fixed: The existing test was deficient, as it was the first `next()` call that raised the error.

This was different in (much) earlier versions of BaseX, in which the query API didn’t cache all results before delivering them to the caller.

Comment (dimitarp) [reply basex-core/src/test/java/org/basex/server/SessionTest.java:556]: yes, I guessed so, but I thought you could know better what is the cause ;)

**Pull request merged**